restli-tools/src/main/java/com/linkedin/restli/tools/snapshot/check/PegasusSchemaSnapshotCompatibilityChecker.java

```diff
@@ -230,6 +276,24 @@ public CompatibilityInfoMap checkPegasusSchemaCompatibility(String prevPegasusSc
       result.getMessages().forEach(message -> _infoMap.addModelInfo(message));
     }
 
+    if (_checkPegasusSchemaAnnotation)
+    {
+      List<SchemaAnnotationHandler.AnnotationCompatibilityResult> annotationCompatibilityResults =
+          AnnotationCompatibilityChecker.checkPegasusSchemaAnnotation(preSchema, currSchema, _handlers);
+      for (SchemaAnnotationHandler.AnnotationCompatibilityResult annotationResult: annotationCompatibilityResults)
+      {
+        if (!annotationResult.getMessages().isEmpty())
+        {
+          annotationResult.getMessages().forEach(message -> _infoMap.addAnnotation(message));
+        }
+      }
+    }
+
+    if (_checkExtensionSchemaAnnotation)
+    {
+      //TODO: extension schema annotation check
```

Comment:
Actually, I am thinking about using a different method in AnnotationCompatibilityChecker for extension schema check.
the extension schema does not require the annotation processor to process the schema, there is no property resolve/override requirements for extension schemas. We just need a SchemaHandler which provides the compatibility check logic for extension schema. Also we need a schema visitor to get pathSpec and properties. It will traverse the schema once, which reduces the unnecessary schema traverse(process annotation).

Reply:
I don't understand how it would be better. You are going to create another branch for extensions which does the. same thing as the annotation checker.

If you have only the extension annotation handler, its visitor would traverse the schema once and get you the same result, right? The path override based traversal will happen only if handlers for annotations that support such overrides are present.

Reply:
Discussed offline.
I introduced a new schema visitor - IdentitySchemaVisitor, which will not traverse the schema. It can be used for extension schema annotation check or other use cases which do not need override properties but would like to do the compatibility check. There will be no extra/unnecessary schema traverse for such case.
Now all the annotation check use cases will fall into the same path: go to annotation processor to process the annotation, then do the annotation compatibility check.